omod/src/main/webapp/resources/scripts/conditionlist/controllers/addcondition.controller.js

```diff
@@ -31,25 +35,46 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
                 emr.errorAlert("Error Saving condition");
             });
         }
+    }
 
     self.initCondition = self.initCondition || function () {
             $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
             $scope.condition = new ConditionModel($scope.patientUuid);
+            self.showEndDate();
         }
-
+        
     self.validateCondition = self.validateCondition || function () {
-            var concept = $scope.concept.concept;
-            if (concept !== null) {
-                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
-            } else {
-                // noncoded diagnosis
-                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
-            }
-
-            $scope.condition.onSetDate = self.getSelectedDate();
+        var concept ;
+        if($scope.concept.concept.uuid)
+        {
+            concept=$scope.concept.concept;
+        }
+        else{
+            concept=$scope.condition.concept;
+        }
+        if (concept !== null) {
+            $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+        } else {
+            // noncoded diagnosis
+            $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+        }
+        $scope.condition.onSetDate = self.getSelectedDate();
+        if($scope.condition.status=='INACTIVE')
```

Comment:
Why are we using 'INACTIVE' in more than one place?

Reply:
Here I'm are checking if condition is INACTIVE or not, to save the value of end date for inactive conditions in the database

Reply:
What happens when we change 'INACTIVE' to 'INNNNACTIVE'?

Reply:
Why would $scope.condition.status become 'INNNNACTIVE' when condition.status takes only two values 'ACTIVE' and 'INACTIVE' ?

Reply:
In how many places would you have to change it?

Reply:
Are you suggesting to use a constant?

Reply:
What do you think?